pkg/monitor/lease.go

```diff
@@ -0,0 +1,137 @@
+package monitor
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/milosgajdos/tenus"
+	"github.com/openshift/baremetal-runtimecfg/pkg/config"
+	"github.com/openshift/baremetal-runtimecfg/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+/*
+	After reading the cmd flags and before executing the keepalived-monitor:
+	1. Get the directory of args[2] = path_to_config (e.g. /etc/keepalived/keepalived.conf) [V]
+	2. Check if the there's a file named "monitor.conf" in that directory. If true, then: [V]
+		2.1. for [API interface, Ingress interface]: . [V]
+			2.1.1 Generate a unique MAC address
+				2.1.1.1. The first 24 bits of the MAC should be the same as the real interface mac address.
+						Get the interface by using `GetInterfaceAndNonVIPAddr()` and after that get its mac address.
+				2.1.1.2. The last 24 bits are generated using some hash function.
+						An example is the way the VRIDs are generated in `PopulateVRIDs()`
+			2.1.2. Create an interface type macvlan per vip using the generated mac address [V]
+			2.1.3. Run dhclient with the new vip interface in the background (goroutine)
```

Comment:
The goroutine part of this seems to be missing.

Reply:
No need for a goroutine. executed with `Start()` function instead of `Run()` which execute a goroutine inside.